JWBestPracticeApps/Captions/Captions/PlayerViewController.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  PlayerViewController.swift
+//  Captions
+//
+//  Created by David Almaguer on 22/11/22.
+//
+
+import UIKit
+import JWPlayerKit
+
+/**
+ Custom protocol to forward caption events.
+ */
+protocol PlayerCaptionsDelegate: AnyObject {
+    func player(_ player: JWPlayer, updatedCaptionList options: [JWMediaSelectionOption])
+    func player(_ player: JWPlayer, captionTrackChanged index: Int)
+    func player(_ player: JWPlayer, captionPresented caption: [String], at time: JWTimeData)
+}
+
+/**
+ The `PlayerViewController` handles the caption events from the player and forward them through the `PlayerCaptionsDelegate`.
+ */
+class PlayerViewController: JWPlayerViewController {
+    weak var captionsDelegate: PlayerCaptionsDelegate?
+    
+    override func jwplayer(_ player: JWPlayer, updatedCaptionList options: [JWMediaSelectionOption]) {
+        super.jwplayer(player, updatedCaptionList: options)
+        DispatchQueue.main.async { [weak self] in
```

Comment:
Overall looks good, the only question I have is if these dispatch calls should be on the JWPVC subclass or the `ViewController`?. It is cleaner if we keep it here, but the object that benefits from these being called on the main thread is the one that will update the UI elements.

Reply:
Good point, I will make the change.